linode_api4/errors.py

```diff
@@ -8,14 +15,90 @@ class ApiError(RuntimeError):
     often, this will be caused by invalid input to the API.
     """
 
-    def __init__(self, message, status=400, json=None):
+    def __init__(
+        self,
+        message: str,
+        status: int = 400,
+        json: Optional[Dict[str, Any]] = None,
+        response: Optional[Response] = None,
+    ):
         super().__init__(message)
+
         self.status = status
         self.json = json
+        self.response = response
+
         self.errors = []
+
         if json and "errors" in json and isinstance(json["errors"], list):
             self.errors = [e["reason"] for e in json["errors"]]
 
+    @classmethod
+    def from_response(
```

Comment:
I originally wanted to implement the formatting logic using a `__str__` override but instead opted for this implementation to reduce the chances of a breaking change.